oauth-client-util/src/main/java/org/radarcns/oauth/OAuth2Client.java

```diff
@@ -0,0 +1,131 @@
+package org.radarcns.oauth;
+
+import okhttp3.Authenticator;
+import okhttp3.Credentials;
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.Route;
+import org.radarcns.exception.TokenException;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class for handling OAuth2 client credentials grant
+ */
+public class OAuth2Client {
+    private URL tokenEndpoint;
+    private String clientId;
+    private String clientSecret;
+    private Set<String> scope;
+    private OAuth2AccessToken currentToken;
+
+    private static OkHttpClient HTTP_CLIENT;
```

Comment:
I understand we have to have to have an `OAuth2Client` for each client credentials. In that case, would it still make sense to have a static `HTTP_CLIENT` that authorizes using credentials from instance variables?

Reply:
Initially i did not have it static. But thanks to @fnobilia for pointing out that `OkHttpClient` is designed to be used this way. It has its own thread and connection pool. So the recommended use is to have only one instance of the class. Reference [here](https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html).